src/components/BookList.jsx

```diff
@@ -1,40 +1,35 @@
 /* eslint-disable linebreak-style */
+// BookList.js
+
 import React from 'react';
-import PropTypes from 'prop-types';
+import { useDispatch, useSelector } from 'react-redux';
+
+function BookList() {
+  const dispatch = useDispatch();
+  const books = useSelector((state) => state.books.books);
 
-function BookList({ books, onRemove }) {
   // Check if books is undefined or empty before rendering
   if (!books || books.length === 0) {
     return <h5>No books available.</h5>;
   }
 
+  const handleRemoveBook = (bookId) => {
+    dispatch({ type: 'REMOVE_BOOK', payload: bookId });
+  };
+
```

Comment:
You need to dispatch the `removeBook` reducer that you created in the **booksSlice.js** file after you import it in this file. Currently this functionality is not working. Kindly fix this issue.